datafusion/functions/src/string/common.rs

```diff
@@ -418,6 +451,88 @@ impl StringArrayBuilder {
     }
 }
 
+pub(crate) struct LargeStringArrayBuilder {
```

Comment:
I wanted to make the already existing `StringArrayBuilder` generic but was having issues 😢